src/sentry/snuba/models.py

```diff
@@ -97,3 +99,26 @@ class Status(Enum):
     class Meta:
         app_label = "sentry"
         db_table = "sentry_querysubscription"
+
+    # We want the `QuerySubscription` to get properly created in Snuba, so we'll run it through the
+    # purpose-built logic for that operation rather than copying the data verbatim. This will result
+    # in an identical duplicate of the `QuerySubscription` model with a unique `subscription_id`.
+    def write_relocation_import(
+        self, pk_map: PrimaryKeyMap, scope: ImportScope, flags: ImportFlags
+    ) -> Optional[Tuple[int, int, ImportKind]]:
+        # TODO(getsentry/team-ospo#190): Prevents a circular import; could probably split up the
+        # source module in such a way that this is no longer an issue.
+        from sentry.snuba.subscriptions import create_snuba_subscription
+
+        old_pk = super()._normalize_before_relocation_import(pk_map, scope, flags)
+        if old_pk is None:
+            return None
+
+        subscription = create_snuba_subscription(self.project, self.type, self.snuba_query)
```

Comment:
Previously, were imports for this working at all? Wondering if there was some setup missing that `create_snuba_subscription` does

Reply:
I don't see how they would have worked. This is one of the (many) ways that old import/export was broken.